Wangscape/tilegen/CartesianPower.h

```diff
@@ -0,0 +1,106 @@
+#pragma once
+#include "CartesianPowerIterator.h"
+#include <utility>
+
+namespace tilegen
+{
+
+template<typename InputIt>
+class CartesianPower
+{
+public:
+    static_assert(std::is_base_of<std::input_iterator_tag, 
+                                  typename std::iterator_traits<InputIt>::iterator_category>::value,
+                  "InputIt must be at least an InputIterator");
+    typedef CartesianPowerIterator<InputIt> const_iterator;
+
+    CartesianPower(InputIt first_, InputIt last_, size_t power_);
+    template<typename Container>
+    CartesianPower(const Container& container, size_t power_);
```

Comment:
Yes, I thought about consistent naming. BTW shouldn't these members (`first`, `last`, `power` be private)?